androidapp/Crisislab/app/src/main/kotlin/com/example/crisislab/MainActivity.kt

```diff
@@ -17,68 +17,80 @@ import okhttp3.Request
 import okhttp3.WebSocket
 
 class MainActivity() : ComponentActivity() {
+    // ViewModels for managing logs and socket status
     lateinit var logViewModel: LogViewModel
-    lateinit var socketStatusViewModel: SocketStatusViewModel;
-    private lateinit var binding:ActivityMainBinding
-    val notificationModule: NotificationModule = NotificationModule;
-    lateinit var notificationHandler: NotificationHandler;
+    lateinit var socketStatusViewModel: SocketStatusViewModel
+    // Binding object for the main activity layout
+    private lateinit var binding: ActivityMainBinding
+    // Notification module instance
+    val notificationModule: NotificationModule = NotificationModule
+    // Notification handler instance
+    lateinit var notificationHandler: NotificationHandler
 
     @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
-        val client: OkHttpClient =  OkHttpClient()
+        val client = OkHttpClient()
+
+        // Initialize ViewModels
         logViewModel = ViewModelProvider(this).get(LogViewModel::class.java)
-        socketStatusViewModel = ViewModelProvider(this).get(SocketStatusViewModel::class.java);
+        socketStatusViewModel = ViewModelProvider(this).get(SocketStatusViewModel::class.java)
         socketStatusViewModel.updateStatus("Status: Disconnected.")
 
-		// If user high enough API version then ask for permission for notifications
+        // Request necessary permissions for notifications
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
-            Log.d("Permission", "Asking for permissions")
-            (ActivityCompat::requestPermissions)(this as Activity, arrayOf(Manifest.permission.POST_NOTIFICATIONS, Manifest.permission.USE_FULL_SCREEN_INTENT), 0)
+            Log.d("Main", "Asking for permissions")
+            ActivityCompat.requestPermissions(this as Activity, arrayOf(Manifest.permission.POST_NOTIFICATIONS, Manifest.permission.USE_FULL_SCREEN_INTENT), 0)
         }
 
+        // Set click listener for connect button
         binding.connect.setOnClickListener {
             connectToSocket(client)
         }
 
+        // Connect to WebSocket and set up RecyclerView
```

Comment:
Hmm I wonder what this function called connectToSocket does (should probably be connectToWebsocket)

Reply:
Fixed, will push soon